src/node_crypto.cc

```diff
@@ -751,6 +751,23 @@ void SecureContext::AddRootCerts(const FunctionCallbackInfo<Value>& args) {
   CHECK_EQ(sc->ca_store_, nullptr);
 
   if (!root_cert_store) {
+#if defined(NODE_SYSTEM_CERT_STORE)
+    // *Assume* OpenSSL is setup correctly, which is the case
+    // for distribution supplied versions.
+    //
+    // If this does not work, define SSL_CERT_DIR environment variable.
+    if (SSL_CTX_set_default_verify_paths(sc->ctx_)) {
+      root_cert_store = SSL_CTX_get_cert_store(sc->ctx_);
+      // root_cert_store created here is already assigned to the SSL_CTX
+      // so when it is assigned again below, the reference is dropped by 1
+      // and then we will delete root store along with the SSL_CTX deletion.
+      // Increase references to 2 to avoid this scenario.
+      CRYPTO_add(&root_cert_store->references, 1, CRYPTO_LOCK_X509_STORE);
+    } else {
+      // Failed to load, default to nothing.
```

Comment:
This creates an undebuggable (by the user of node) situation. Is it not possible to print some kind of error here? What are the possible reasons for failure? non-existence of cert store? invalid format of cert store? ...?